Modules/_abc.c

```diff
@@ -128,31 +219,29 @@ _in_weak_set(PyObject *set, PyObject *obj)
 static int
 _in_cache(PyObject *self, PyObject *cls)
 {
-    PyObject *key, *cache;
-    key = PyWeakref_NewRef(self, NULL);
-    if (!key) {
-        return -1;
-    }
-    cache = PyObject_GetItem(_the_cache, key);
-    if (!cache) {
+    _abc_data *impl;
+    int res;
+    impl = (_abc_data *)_PyObject_GetAttrId(self, &PyId__abc_impl);
```

Comment:
_PyObject_GetAttrId doesn't necessarily return an _abc_data

(the same problem occurs throughout the file, only mentioning this once to avoid duplicate comments)